src/Superpower/Parsers/Span.cs

```diff
@@ -258,5 +258,33 @@ public static TextParser<TextSpan> MatchedBy<T>(TextParser<T> parser)
                     result.Remainder);
             };
         }
+
+        /// <summary>
+        /// Parse as much of the input as matches <paramref name="text" />.
+        /// </summary>
+        /// <param name="text">The text to match until. The text is not included in the result.</param>
+        /// <returns>A parser that will match everything until the text argument is matched.</returns>
+        /// <exception cref="ArgumentNullException">The text is null.</exception>
+        public static TextParser<TextSpan> Until(string text)
+        {
+            if (text == null) throw new ArgumentNullException(nameof(text));
+
+            bool isWithinLength(TextSpan ts) => ts.Length >= text.Length;
+            bool isStopwordMatching(TextSpan ts) => ts.First(text.Length).EqualsValue(text);
+            bool isMatch(TextSpan ts) => isWithinLength(ts) && isStopwordMatching(ts);
+
+            return (TextSpan input) =>
+            {
+                TextSpan x = input;
+
+                while (!x.IsAtEnd)
```

Comment:
It may be nice to look at using `string.IndexOf` here to speed up the search, since it'll be cleverly implemented at the low level? There's a similar algorithm in the regex matcher at: https://github.com/datalust/superpower/blob/dev/src/Superpower/Parsers/Span.cs#L219, I think.

Reply:
Not sure I understood the indexOf reference you link to - is it the right place?
Anyway, I made a new version that I believe should be faster, as it only scans the input once.
Maybe it's better that you do the implementation, I think you are far more into all the details.